packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -112,7 +113,8 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
               aria-haspopup='true'
               autoCapitalize='off'
               autoComplete='off'
-              role='combobox' />
+              role='combobox'
+              disabled={ this.state.items.length >= itemLimit } />
```

Comment:
Wouldn't it be possible to simply disable the underlying AutoFill component instead of the entire Picker component? 
This would allow you to still remove existing items to re-enable the AutoFill.